### DISCUSSION TITLE
Anzeige des Diagramms fehlerhaft

### DISCUSSION THREAD

**Discussion by chef76** [brutella/hkknx-public#339] [General]
Die Anzeige des Leistungsdiagramms ist fehlerhaft:
![image](https://github.com/user-attachments/assets/c307bd65-382b-4b09-b5f1-eb0330a418b5)
Das Diagramm hat eine Skala von 0 -4 und eine Spitze bei 3,2. Die letzten Werte liegen aber bei 8 - 23 Watt

**Comment by brutella** [comment]
Ich vermute mal, dass für den Zeitraum 9:40 - 9:50 so einen niedrigen Mittelwert berechnet wurde, und deshalb die Spitzen nicht dargestellt werden. Ist die Anzeige korrekt, wenn du als Zeitraum "Letzte Stunde" oder "Letzten 6 Stunden" auswählst?

**Reply by chef76** [reply]
Die letzte Stunde ist leider schon rum, bei den letzten 6 sieht es aber auch nicht besser aus:
![image](https://github.com/user-attachments/assets/31d42ed1-acf0-426e-b946-5fe43f012783)
Wenn Du Mittelwerte anzeigst, erklärt es das natürlich, macht aber keinen Sinn. Ich schlage vor Du zeigst Max. Werte an, wie es alle anderen Diagrammtools auch machen. Wenn man so ein Diagramm anschaut, will man Max.-Werte sehen und keine Mittelwerte.

**Reply by brutella** [reply]
Trotzdem vermute ich, dass die schnellen Ausschläge (bis zu 23W innerhalb von ein paar Sekunden) durch den Mittelwert nicht dargestellt werden. Werd versuchen die Anzeige dafür anzupassen.

**Reply by chef76** [reply]
![image](https://github.com/user-attachments/assets/800baf27-f4a1-4059-8e65-580791c9554b)

![image](https://github.com/user-attachments/assets/52643019-be39-4194-aa92-afc5e906cc4b)

Hier noch mal der Unterschied zwischen real (oben) und deinem Diagramm 😞

**Comment by brutella** [comment]
Hierzu gibt es in Version [3.0.0-b1](https://github.com/brutella/hkknx-public/releases/tag/3.1.0-b1) etwas Neues. Einerseits wurde die Auflösung für Diagrammen bei kurzen Zeiträumen (24h) erhöht, andererseits kann man mit der Option "Min./Max.-Bereich" die minimalen und maximalen Werte anzeigen lassen.

Ist damit dein Problem gelöst?

**Reply by chef76** [reply]
Ich habe auf die aktuelle Version upgedated und die alten Monitoringdaten gelöscht. Trotz mehrfachem Neustart zeigt keines der Diagramme mehr was an. 😩

**Reply by brutella** [reply]
`die alten Monitoringdaten gelöscht` WHY? 😭

**Reply by chef76** [reply]
Weil nichts in den Diagrammen angezeigt wurde und noch immer nicht wird!!?

**Reply by chef76** [reply]
Auch mit der aktuellsten Version sind die Diagramme alle leer, obwohl seit gestern 8,4 MB an Daten gespeichert wurden.

**Reply by brutella** [reply]
Sind auch die Diagramme von KNX Gruppenadressen leer?

**Reply by chef76** [reply]
Ja, auch die Diagramme der Gruppenadressen sind leer:

![image](https://github.com/user-attachments/assets/3f39f157-fcc3-43e9-8796-d4a26ee528e1)

**Reply by brutella** [reply]
@chef76 Wird unter "Letzter Wert" etwas angezeigt? Ist auch ein Datentyp für diese Gruppenadresse definiert?

**Reply by chef76** [reply]
Ja und der datentyp ist definiert. Bis gestern vor dem Update wurde das Diagramm auch noch angezeigt

![image](https://github.com/user-attachments/assets/6b76f9f5-6ec4-433f-8294-26c9106179d4)

**Reply by brutella** [reply]
Merkwürdig 🤨 Auf meinem System und auch in der [Demo](https://hkknx.hochgatterer.me/projects/cotlava698vmorb3og3g/monitoring/charts?ctype=security) werden die Diagramme unter Version 3.1.0-b3 korrekt angezeigt.

Werden unter *Monitoring → System* die Diagramme für CPU- und Busauslastung angezeigt?

**Reply by chef76** [reply]
Ja, alles normal: 
![image](https://github.com/user-attachments/assets/b9ffbd3f-b785-4490-8ea8-c6c20ab8c928)

**Comment by brutella** [comment]
Ist deine Systemzeit korrekt?
Hast du dein System schonmal neu gestartet?

**Reply by brutella** [reply]
Kannst du bitte den Web Inspector (im Safari mit `Option + Cmd + I`) öffnen und nachsehen, ob irgendwelche Fehler im Log stehen?

**Reply by chef76** [reply]
Hier mal die Ausgabe von Safari: 
<img width="1904" alt="Bildschirmfoto 2024-10-06 um 17 04 13" src="https://github.com/user-attachments/assets/bf9b4dca-3d24-4c9f-8ca9-860ac8ce065a">

[a comment by an automated account is not shown]

**Comment by brutella** [comment]
Bitte Version [3.1.0-b3](https://github.com/brutella/hkknx-public/releases/tag/3.1.0-b3) ausprobieren und falls immer noch Probleme auftauchen im Systemprotokoll nach Fehlern Ausschau halten. Danke!

**Reply by chef76** [reply]
Auch bei mir mit der neuen Version keine Daten in den Diagrammen

**Reply by chef76** [reply]
![image](https://github.com/user-attachments/assets/43a38fa1-f855-48bc-9c37-cac5fc975da1)

**Reply by brutella** [reply]
Findet ihr im Systemprotokoll Hinweise zu einem Fehler?

**Reply by chef76** [reply]
Leider nicht. Nichts Ungewöhliches im Systemlog.

**Reply by brutella** [reply]
Blöde Frage aber ist eure Systemzeit korrekt?

**Reply by chef76** [reply]
NTP synchronisiert: ![image](https://github.com/user-attachments/assets/ad38ba63-4fee-443b-ab6b-634294a4dccf)

**Reply by chef76** [reply]
... auf den Clients natürlich auch!

**Reply by brutella** [reply]
@hr-automation Kannst du mal im Chrome die 3 Fehler auf der Seite genauer ansehen? Fehler habe ich nämlich bei mir nicht.

**Reply by brutella** [reply]
Sorry, hab das hier gemeint. ![image](https://github.com/user-attachments/assets/0d72958e-1ec1-47b4-8b71-f9721ad5e81a)

[a comment by an automated account is not shown]

**Reply by brutella** [reply]
- Wird im Log (Fußzeile → Systemprotokoll) ein Fehler angezeigt, wenn du ein Diagramm öffnest?
- Wird keines der Diagramme angezeigt, also auch die unter Monitoring → Licht/Klima/Sicherheit, oder nur bestimmte Diagramme?

**Reply by brutella** [reply]
Hast du während der Aufzeichnung der Log-Ausgaben ein Diagramm geöffnet? Kann dazu keine Log-Ausgsabe finden.

**Reply by brutella** [reply]
Werden unter `...` anderen Datenquellen angezeigt? Kannst du diese auswählen und anzeigen lassen?

**Reply by brutella** [reply]
Alles sehr merkwürdig. 🤨
Könntest du mir bitte ein Backup (Einstellungen → Mehr → Datensicherung erstellen) per E-Mail hkknx@hochgatterer.me zukommen lassen? Danke.

**Reply by brutella** [reply]
@hr-automation Vielen Dank für die Daten. Jetzt konnte feststellen, dass in deinem Fall die Datenbank für die Monitoring-Daten korrupt war. Warum das genau passiert ist, kann ich nicht genau sagen – womöglich ein Stromausfall? Deshalb konnten keine Monitoring-Daten angezeigt werden.

In Version [3.1.0-b4](https://github.com/brutella/hkknx-public/releases/tag/3.1.0-b4) wird dieser Fehler behandelt und die Datenbank wird automatisch "repariert".

**Reply by chef76** [reply]
Bei mir bringt die 3.1.0-b4 leider keine Verbesserung. Bei allen Diagrammen stehen letzte Werte, es werden aber keine Diagramme angezeigt. 🤔

**Reply by brutella** [reply]
Könntest du mir bitte auch ein Datensicherung zur Verfügung stellen? Eine E-Mail an hkknx@hochgatterer.me mit der Datei dazu wär super!

**Reply by brutella** [reply]
@chef76 Vielen Dank für deine Daten. Ich konnte mit Hilfe deiner Datensicherung die Diagramme bei mir anzeigen lassen – musste die Anzeigen auf "Letzten 7 Tage" ändern. Kann es sein, dass du keine Verbindung zu deiner KNX Installation hast und deshalb keine Diagramme angezeigt werden?

Wie auch immer, mit Version [3.1.0-b5](https://github.com/brutella/hkknx-public/releases/tag/3.1.0-b5) konnte ich Probleme mit der Sicheren Verbindung auf dem Raspberry Pi behoben und damit auch die Daten von @chef76 korrekt anzeigen.

**Reply by chef76** [reply]
@brutella Kann es heute nicht mehr testen. Der Kontakt zu meiner KNX-Installation ist völlig in Ordnung. Was hat das mit der Anzeige der Diagramme zu tun, wenn die Daten eh in der Datenbank liegen? Außerdem musst du doch an den Zeitstempeln der Daten sehen, dass die bis zur Erstellung des Backup da sind?!?

**Reply by brutella** [reply]
> Außerdem musst du doch an den Zeitstempeln der Daten sehen, dass die bis zur Erstellung des Backup da sind?!?

Stimmt, die Daten sind vorhanden.

**Reply by chef76** [reply]
> @brutella Feedback: Trotz neuer Monitoring Datenbank - Wiederherstellung / Update auf die b5 keine Veränderung!

Auch bei mir mit -b5 keine Änderung 😭

**Reply by brutella** [reply]
Hab ich schonmal versucht die Version 3.0.0 zu installieren? Funktioniert es damit wieder?

**Reply by brutella** [reply]
Funktioniert es auch, wenn du jetzt wieder die aktuellste Beta installiert? (Es könnte sein, dass sich bei den Einstellungen in einer der Beta-Versionen ein Fehler eingeschlichen hat, und deshalb die Diagramme nicht mehr funktioniert haben.)

**Reply by brutella** [reply]
@hr-automation Ich vermute, du verwendest das "alte" Raspberry Pi Image, oder? Wenn ja, dann musst du sicherstellen, dass das Datum korrekt gesetzt wird. Im neuen Raspberry Pi Image wird das zuverlässiger gemacht als wie im alten Image.

Wird das Diagramme korrekt angezeigt, wenn du einen fixen Zeitraum auswählst?

**Reply by brutella** [reply]
War leider die ganze letzte Woche mit der UI für die Farbauswahl für Lampen beschäftigt (ist in [3.1.0-b6](https://github.com/brutella/hkknx-public/releases/tag/3.1.0-b6) verfügbar) und werd mich morgen wieder um das Problem kümmern können.

**Reply by brutella** [reply]
Je mehr über dieses Problem nachdenke, desto mehr glaube ich, dass es ein Problem mit der Systemzeit ist. 
Wird im Diagramm etwas angezeigt, wenn du den Zeitraum (Start- und Enddatum) selbst festlegst?

@hr-automation  Hast du das "alte" Raspberry Pi Disk Image in Verwendung oder hast du hkknx manuell installiert?

**Reply by chef76** [reply]
@brutella : Denkanstoß- du hast von uns beiden Backups (übrigens nutze ich auch das alte PI Image erweitert um NTP Dämon). Da siehst du doch an den Daten die Zeitstempel. Wenn die alle korrekt sind, wo kommt die Zeit her, die für die Anzeige verwendet wird?

**Reply by brutella** [reply]
Ja, es scheint irgendwas mit der Systemzeit nicht zu stimmen.

In Version [3.1.0-b8](https://github.com/brutella/hkknx-public/releases/tag/3.1.0-b8) wird jetzt unter *Monitoring → Systemstatus* explizit die Systemzeit angezeigt. Wenn man mit dem Mauszeiger über die Zeit fährt, wird Information über die Zeitzonen dargestellt. Welche Informationen scheinen da bei euch auf?

**Reply by chef76** [reply]
![image](https://github.com/user-attachments/assets/ed172e72-9ede-4b92-8ad7-9545fffd706c)

**Reply by chef76** [reply]
Ich hab eben auch mal das 3.0.0 Image auf einen neuen PI 5 installiert und mein Backup eingespielt. Prima Arbeit, hat alles 1A funktioniert. 

Nach dem Start allerdings zwar 55 MB Diagrammdaten, aber keine Anzeige. Daten löschen und sofort wurden die ersten Werte im Diagramm wieder angezeigt. 😒
Da ich das aber nur als Versuch sehe, würde ich schon bevorzugen, dass das auf der alten Installation auch funktioniert. Wundert mich, dass sich da außer uns beiden sonst noch niemand beschwert hat.

**Reply by EugHel** [reply]
Bei mir wird es so angezeigt. v3.1.0-b8
![image](https://github.com/user-attachments/assets/7cacfea2-5fbc-4dbc-a12f-2df5f475b835)

**Reply by EugHel** [reply]
Sorry, aber ich komme bei der Zeit nicht weiter...

In der Synology ist die richtige Zeit eingestellt:
![image](https://github.com/user-attachments/assets/0cc962bb-043a-466b-96fc-e65515d512fb)
Im Container wird es so angezeigt:
![image](https://github.com/user-attachments/assets/203d09eb-7f94-4a52-9ed7-317582d7b2cf)

Wieso wird die Systemzeit bei mir im hkknx UTC +0 angezeigt?

Achja. Alles neu gestartet habe ich bereits.

[a comment by an automated account is not shown]

**Comment by brutella** [comment]
@chef76 @hr-automation Bei euch scheint die korrekte Zeit eingestellt zu sein. Bei den Diagrammen müsste also die Zeit passen – könnt ihr das bestätigen, dass wenn ihr als Zeitraum "Letzte Woche" auswählt, dass auch die richtigen Tage angezeigt werden?

@EugHel Bei dir ist wohl die Zeitzone falsch. Bei aktueller Winterzeit sollte +1h angezeigt werden.

**Reply by chef76** [reply]
Die liegt wohl das Problem:

![image](https://github.com/user-attachments/assets/1f7abb79-ef87-40e8-a389-42d3ed5fc2ec)
Diese Woche ist nicht 23. - 29. Dezember!

**Reply by chef76** [reply]
Und die Systemzeit passt: 
![image](https://github.com/user-attachments/assets/0bb9b3a7-1427-4542-ab1d-82d2df2fa8cf)

**Comment by brutella** [comment]
Was wird angezeigt, wenn du für das Diagram ein Start- und Enddatum festlegst? zB Start: 20/10/2024, Ende: 29/10/2024

**Reply by chef76** [reply]
Dumme Rückfrage: wie stelle ich einen benutzerdefinierten Zeitraum ein? Ich kam. Nur die festen auswählen:![image](https://github.com/user-attachments/assets/07129f2b-64e8-472f-b3b0-4ecc0a2f57d6)

**Reply by brutella** [reply]
Unter den vordefinierten Zeiträumen, kann ein Start- und Enddatum festgelegt werden. Ist auf dem iPhone etwas schwer ersichtlich.
<img width="476" alt="Screenshot 2024-10-29 at 12 28 37" src="https://github.com/user-attachments/assets/6f4b4a0f-14fb-4968-98a8-972c7965706a">

**Reply by chef76** [reply]
Das Diagramm ignoriert die Eingabe: ![image](https://github.com/user-attachments/assets/19f3707d-d432-4403-b2b8-52aed3ca0e65)

**Reply by brutella** [reply]
Mit Version [3.1.0-b9.1](https://github.com/brutella/hkknx-public/releases/tag/3.1.0-b9.1) sollte das Problem nun endgültig behoben sein. Das Problem war ein Integer-Overflow bei 32-Bit Systemen.

**Reply by guyyannick** [reply]
Jaaaa, es funkt wieder 🚀

**Reply by chef76** [reply]
Kann ich bestätigen. Auch bei mir wieder alles gut! Das war eine schwere Geburt. 😩

Danke!

[a comment by an automated account is not shown]
